### PR TITLE
Add Harmony Presence

### PR DESCRIPTION
**Pull request by marinofranz**
image

### CODE REVIEW

File: websites/H/Harmony/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "Harmony",
+    "author": {
+      "name": "marino!",
+      "id": "598953747963707409"
+    },
+    "category": "music",
+    "logo": "https://talk.weareharmony.net/branding/default.png",
+    "thumbnail": "https://talk.weareharmony.net/branding/background-nologo.png",
```
Comment (EncryptedDev) [inline websites/H/Harmony/dist/metadata.json:10]: logo and thumbnail must be on imgur, read the guidelines

File: websites/H/Harmony/presence.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logobg",
+    smallImageKey: "nobg_white"
+  };
+
+  if(
+    document.location.hostname === "weareharmony.net" &&
+    document.location.hash.startsWith("#/")
+  ) {
```
Comment (EncryptedDev) [inline websites/H/Harmony/presence.ts:34]: handled automatically, not needed
```suggestion
  if(document.location.hash.startsWith("#/")) {
```

File: websites/H/Harmony/presence.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
+const presence = new Presence({
+    clientId: "882003722270572574"
+  }),
+  browseTimestamp = Math.floor(Date.now() / 1000);
+
+let npTitle = "WeAreHarmony.net",
+  npArtist = "We play the biggest hits!",
+  npOnAir = "AutoDJ",
+  npListeners = 0;
```
Comment (EncryptedDev) [inline websites/H/Harmony/presence.ts:9]: What's the point of pre-assigning this?

File: websites/H/Harmony/presence.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    smallImageKey: "nobg_white"
+  };
+
+  if(
+    document.location.hostname === "weareharmony.net" &&
+    document.location.hash.startsWith("#/")
+  ) {
+    presenceData.startTimestamp = browseTimestamp;
+    presenceData.smallImageText = `Listeners: ${npListeners} | Live DJ: ${npOnAir}`;
+
```
Comment (EncryptedDev) [inline websites/H/Harmony/presence.ts:37]: empty line
```suggestion
```

File: websites/H/Harmony/presence.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      presenceData.state = `💿 ➜ ${npOnAir}`;
+    }
+  }
+
+  if(presenceData.details === null) {
+    presence.setTrayTitle();
+    presence.setActivity();
+  } else 
+    presence.setActivity(presenceData);
+
```
Comment (EncryptedDev) [inline websites/H/Harmony/presence.ts:70]:
```suggestion
    presence.setActivity(presenceData);
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
prettify your code too

**Comment by marinofranz** [comment]
I've turned off my computer right now, so at the moment I am unable to

**Review by Slowlife01 — DISMISSED** [review]

File: websites/H/Harmony/dist/metadata.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "Harmony",
+    "author": {
+      "name": "marino!",
+      "id": "598953747963707409"
+    },
+    "category": "music",
+    "logo": "https://i.imgur.com/VCFXbxY.png",
+    "thumbnail": "https://i.imgur.com/7nZITkG.png",
```
Comment (Bas950) [inline websites/H/Harmony/dist/metadata.json:10]: must be a promotional banner or a screenshot of the website homepage.

**Review by Bas950 — CHANGES_REQUESTED** [review]
> Presences for online radios are only allowed if the radio has at least 100 weekly listeners and 15 concurrent

Upload proof of Harmony meeting this requirement.

**Comment by marinofranz** [comment]
> > Presences for online radios are only allowed if the radio has at least 100 weekly listeners and 15 concurrent
> 
> Upload proof of Harmony meeting this requirement.

https://i.imgur.com/Y4uOnmH.png
I was hoping, since we are very under the limit, you could make a small exception? During the most popular times when DJs are on air, we gain around 13-17 listeners.

**Comment by EncryptedDev** [comment]
I've checked every day over the last three days and can confirm Harmony does not reach the concurrent listener count

Try again once Harmony has surpassed the requirements as it is not evident Harmony does currently

**Comment by marinofranz** [comment]
> I've checked every day over the last three days and can confirm Harmony does not reach the concurrent listener count
> 
> Try again once Harmony has surpassed the requirements as it is not evident Harmony does currently

Alright, my apologies
I'll just mark this as a draft and mark it ready for review when we reach the listener count

Thanks

**Comment by Slowlife01** [comment]
No draft just open another pull request when it's met the requirements

**Pull request closed without merging**